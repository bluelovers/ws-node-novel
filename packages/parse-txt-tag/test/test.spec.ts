import { join } from "path";
import { readFile } from 'fs-extra';
import { parse } from '..';

test(`parse`, async () =>
{
	const rootDir = join(__dirname, '..');

	let actual = await readFile(join(rootDir, 'test/res', '排版格式.txt'))
		.then(buf =>
		{
			return parse(buf.toString(), {
				on: {
					img(data)
					{
						let { tagName, innerContext } = data;

						console.dir({
							tagName,
							innerContext,
						});

						return null // 如果回傳非 null 則會取代原始文字內容
					},
					default({
						tagName,
						innerContext,
						cache,
						attach,
					})
					{
						console.dir({
							tagName,
							innerContext,
						});

						return null // 如果回傳非 null 則會取代原始文字內容
					},
				},
			})
		})
	;

	expect(actual).toMatchSnapshot();

});
