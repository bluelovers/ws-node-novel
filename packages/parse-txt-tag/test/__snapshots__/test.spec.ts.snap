// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse 1`] = `
Object {
  "attach": Object {
    "images": Object {},
  },
  "cache": Object {},
  "context": "
<ruby>漢<rp>(</rp><rt>Kan</rt><rp>)</rp>字<rp>(</rp><rt>ji</rt><rp>)</rp></ruby>

<ruby>漢<rt>Kan</rt>字<rt>ji</rt></ruby>

<s>刪除綫</s>
<b>粗</b>
<i>斜</i>

这段文本包含 <sup>上标</sup>
这段文本包含 <sub>下标</sub>

---------------------

以下是以全形來使用排版語法

<ruby>漢<rp>（</rp><rt>Ｋａｎ</rt><rp>）</rp>字<rp>（</rp><rt>ｊｉ</rt><rp>）</rp></ruby>

<ruby>漢<rt>Ｋａｎ</rt>字<rt>ｊｉ</rt></ruby>

<s>刪除綫</s>
<b>粗</b>
<i>斜</i>

这段文本包含　<sup>上标</sup>
这段文本包含　<sub>下标</sub>

---------------------

靜讀天下

(圖片html001)
",
}
`;
